.github/workflows/ci.yml

```diff
@@ -72,6 +71,17 @@ jobs:
           cd ..
           echo "LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$PWD/usr/lib" >> $GITHUB_ENV
           echo "TOOLCHAIN_OPTION=$TOOLCHAIN_OPTION -DCMAKE_PREFIX_PATH=$PWD/usr" >> $GITHUB_ENV
+      - name: Prepare OpenSSH Server Container for Testing
```

Comment:
If this there it's own task, it could be done just once for the whole matrix.

Reply:
For the container to actually be available during the build run we would still need to perform a pull/build action for each build run. We could perform a separate task to build and cache it once to prevent the first few builds from duplicating effort, but this case will only come up when we have an empty cache (this PR) or if anything ever changes the Dockerfile or it's build context, which should be fairly rare.

Reply:
You are right, it would only actually do something the first time or if the Dockerfile changes, but the matrix builds then only need a pull, not a build. ;-)

Reply:
It would still need the build for at the very least this PR as due to the GitHub security policies. As mentioned in the top comment, PRs from forked repos (such as this PR) will not be able to push a container to the repo. This means that for this PR where we don't have a container already cached the separate task would not be able push the container it built and the build tasks would have nothing to pull, necessitating a build.

https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token